build-infer.sh

```diff
@@ -96,12 +98,21 @@ check_installed () {
 }
 
 setup_opam () {
-    OCAML_VERSION="4.02.3"
-
     opam init -j $NCPU --no-setup --yes
 
-    OPAMSWITCH=infer-$OCAML_VERSION
-    opam switch install -j $NCPU $OPAMSWITCH --alias-of $OCAML_VERSION || true
+    # get the present ocaml version, the idea here is to avoid installing ocaml compiler if the
+    # default is $SUPP_OCAML_VERSION.
+    present=$(opam config var ocaml-version)
+    if [[ $present == $SUPP_OCAML_VERSION ]];then
+        # get the switch corresponding the version
+        ALIAS_SWITCH=$(opam config var switch)
+    else
+        # if the default is not $SUPP_OCAML_VERSION, download ocaml compiler corresponding to it
+        ALIAS_SWITCH=$SUPP_OCAML_VERSION
+    fi
+
+    OPAMSWITCH=infer-$SUPP_OCAML_VERSION
+    opam switch install -j $NCPU $OPAMSWITCH --alias-of $ALIAS_SWITCH || true
```

Comment:
Using `--alias-of $(opam config var ocaml-version)` is the same as the code before this PR.  I no longer understand how the original goal of this PR can be achieved.  Passing `--alias-of <version>` when `<version>` is already installed still fetches and builds ocaml in all scenarios I have tested.  @ronin13, what am I overlooking?